### DISCUSSION TITLE
Does bleak support ble connection security?

### DISCUSSION THREAD

**Discussion by snowuyl** [hbldh/bleak#863] [Q&A]
* bleak version: 0.14.3
* Python version: 3.10.0
* Operating System: Windows 10

### Description

I would like to make secure connection with Nordic nRF52832. Do you have suggestions?

**Comment by dlech** [comment]
This is mostly handled automatically by the OS. Have a look at the `BleakClient.pair()` method and the open pull requests.

**Reply by snowuyl** [reply]
Form bleak document 0.14.3. It seems bleak doesn't support authentication (EncryptionAndAuthen- tication - Pair the device using encryption and authentication. (This will not work in Bleak...). Does bleak support authentication?
The following description citing from bleak document 0.14.3.
pair(protection_level: int = None, **kwargs) → bool Attempts to pair with the device.
Keyword Arguments protection_level –
DevicePairingProtectionLevel 1: None - Pair the device using no levels of protection. 2: Encryption - Pair the device using encryption. 3: EncryptionAndAuthen- tication - Pair the device using encryption and authentication. (This will not work in Bleak...)

**Comment by snowuyl** [comment]
Does bleak provide api for input passkey?

**Reply by snowuyl** [reply]
For example, passkey is “123456”.

**Reply by bojanpotocnik** [reply]
There is https://github.com/hbldh/bleak/pull/640 open, but I am still trying to find some free time to apply all the comments and update it for the newest version of Bleak.

**Comment by snowuyl** [comment]
Thanks your efforts!
